projects/packages/videopress/src/client/components/video-upload-area/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { Button, useBreakpointMatch, Text } from '@automattic/jetpack-components';
+import { Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { Icon, cloudUpload } from '@wordpress/icons';
+import classnames from 'classnames';
+import { DragEvent, useCallback, useState } from 'react';
+import styles from './style.module.scss';
+import { VideoUploadAreaProps } from './types';
+import type React from 'react';
+
+/**
+ * Video Upload Area component
+ *
+ * @param {VideoUploadAreaProps} props - Component props.
+ * @returns {React.ReactNode} - VideoUploadArea react component.
+ */
+const VideoUploadArea: React.FC< VideoUploadAreaProps > = ( {
+	isLoading = false,
+	className,
+	onSelectFile,
+} ) => {
+	const [ isSm ] = useBreakpointMatch( 'sm' );
+	const [ isDraggingOver, setIsDraggingOver ] = useState( false );
+
+	const handleClickEvent = useCallback( () => {
+		const input = document.createElement( 'input' );
+		input.type = 'file';
+
+		input.addEventListener( 'change', ( e: Event ) => {
+			const target = e.target as HTMLInputElement;
+			onSelectFile( target.files[ 0 ] );
+			input.remove();
+		} );
+
+		input.click();
+	}, [ onSelectFile ] );
```

Comment:
Instead of dynamically inserting a new `input` and clicking on it, I suggest inserting a `input` at the `html` code, making it `absolute`, and taking the entire `width` and `height.`

This provides such a better experience for a11y, it already prevents the user from dragging multiple files in a better UX.

Below is the difference between both:

_Using input_

https://user-images.githubusercontent.com/1663717/184714543-95a97fcf-0afa-4e8e-8ad7-3f830feb032c.mp4

_Dinamically_


https://user-images.githubusercontent.com/1663717/184714607-6881807d-26fc-4f89-b885-df7f94960405.mp4

Reply:
Having the input over the wrapper did not work, as that impedes the user from clicking on the button, and if it is not above the wrapper or with `pointer-events: none`, the hover events don't fire.
What I did, though, was keep the hidden file input with `display: none`. Let me know what you think.